cpp/models/lct_secir/infection_state.h

```diff
@@ -0,0 +1,247 @@
+     * If the index does not match an infectionstate, the return value will be -1.  
+     * @return Index of the first Subcompartment for a vector with one entry per subcompartment or -1.
+     */
+    int get_firstindex(int infectionstatebaseindex) const
+    {
+        if ((0 <= infectionstatebaseindex) && (infectionstatebaseindex < (int)InfectionStateBase::Count)) {
+            return m_subcompartment_indexfirst[infectionstatebaseindex];
+        }
+        else {
+            return -1;
+        }
+    }
+
+    /**
+     * @brief Gets the total number of (sub-)compartments of infection states.
+     */
+    int get_count() const
+    {
+        return m_count;
+    }
+
+    /**
+     * @brief Checks constraints on infection states.
+     *
+     * @return Returns true if one (or more) constraint(s) are not satisfied, otherwise false.
+     */
+    bool check_constraints() const
+    {
+        if (!(m_subcompartment_numbers.size() == (int)InfectionStateBase::Count)) {
+            log_error("Vector for number of subcompartments has the wrong size.");
+            return true;
+        }
+        if (!(m_subcompartment_numbers[(int)InfectionStateBase::Susceptible] == 1)) {
+            log_error("Susceptible compartment can not have subcompartments.");
+            return true;
+        }
+        if (!(m_subcompartment_numbers[(int)InfectionStateBase::Recovered] == 1)) {
+            log_error("Recovered compartment can not have subcompartments.");
+            return true;
+        }
+        if (!(m_subcompartment_numbers[(int)InfectionStateBase::Dead] == 1)) {
+            log_error("Dead compartment can not have Subcompartments.");
+            return true;
+        }
+        for (int i = 0; i < (int)InfectionStateBase::Count; ++i) {
+            if (m_subcompartment_numbers[i] < 1) {
+                log_error("All compartments should have at least one Subcompartment.");
+                return true;
+            }
+        }
+        return false;
+    }
+
+private:
+    /**
+     * @brief Calculates Index of the first Subcompartment for a vector with one entry per subcompartment.
+     *
+     * Therefore the vector with number of subcompartments per infection state is used.
+     */
+    void set_compartment_index()
+    {
+        int index = 0;
+        for (int i = 0; i < (int)(InfectionStateBase::Count); i++) {
+            m_subcompartment_indexfirst[i] = index;
+            index                          = index + m_subcompartment_numbers[i];
+        }
+        m_count = index;
+    }
+
+    std::vector<int>
+        m_subcompartment_numbers; ///< Vector which defines the number of Subcompartments for each infection state of InfectionStateBase.
+    std::vector<int>
+        m_subcompartment_indexfirst; ///< Vector with Indexes for all infection states of the first Subcompartment for a vector with one entry per subcompartment.
+    int m_count; ///< Total number of (sub-)compartments of infection states.
```

Comment:
Is m_count the number of subcompartments or the number of compartments? The way you use it above it is the number of subcompartments but with this description it is unclear to me what is meant.

Reply:
I think its better now.